### ISSUE TITLE
False RCS1229

### ISSUE THREAD

**Issue by NN---** [JosefPihrt/Roslynator#821]
**Product and Version Used**: 
Roslynator.Analyzers 3.2.0

**Steps to Reproduce**:

```cs
using System;
using System.Threading;
using System.Threading.Tasks;

[assembly:CLSCompliant(false)]

namespace ConsoleApp1
{
    public static class Program
    {
        private static ValueTask<int> ReceiveAsync(byte[] data, CancellationToken token)
        {
            using var cancellations = CancellationTokenSource.CreateLinkedTokenSource(token, token);

            return ValueTask.FromResult(data.Length);
        }

        public static void Main()
        {
        }
    }
}
```

**Actual Behavior**:
```csharp
Severity	Code	Description	Project	File	Line	Suppression State
Message	RCS1229	Use async/await when necessary.	

```

**Expected Behavior**:
No message

**Comment by JosefPihrt** [comment]
Hi, I'm not able to reproduce described behavior with the exact same code, no diagnostic is reported.

**Comment by NN---** [comment]
Fixed sample.
It needed few more lines.

**Comment by JosefPihrt** [comment]
I think that analyzer is right in this case. For further explanation see "[The curious case of async, await, and IDisposable](http://thebillwagner.com/Blog/Item/2017-05-03-ThecuriouscaseofasyncawaitandIDisposable)"

**Comment by NN---** [comment]
The method is totally synchronous , why do I need to make it async and have performance degradation ?

**Comment by JosefPihrt** [comment]
Oh, you mean that in the case where `Task.FromResult` is used the diagnostic should not be reported?

**Comment by NN---** [comment]
Yes.
There is no point to make method async if there is direct return using Value/Task.FromResult/Cancellation/Exception

**Comment by JosefPihrt** [comment]
That makes sense.
